### ISSUE TITLE
[$250] Chat - "From" link from IOU details system message thread leads to transaction details twice

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#42221]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.74-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue found when executing PR:** https://github.com/Expensify/App/pull/40983
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Log in to New Expensify
2. Navigate to chat with no transaction between users
3. Submit expense and change its Date
4. Hover over "changed the date" system message and click on Reply in thread
5. Click on From link in the header twice

## Expected Result:
After clicking on From link twice user should return to the main chat.

## Actual Result:
User is directed to the IOU details page twice when clicking on from link. First to the "[amount] expense", then to the "[user] owes [amount]".
User is forced to click on from link thrice to return to main chat.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/9c514de1-73f0-42c7-84fd-9a167504209f

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b763c90751e392a8</li>
        <li>Upwork Job ID: 1808668907625708205</li>
        <li>Last Price Increase: 2024-07-04</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to  #wave-collect - Release 1n

**Comment by kbecciv** [comment]
@mallenexpensify FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by mallenexpensify** [comment]
Unable to reproduce.  Edge case issue too.  Throwing retest weekly and needs repro on this, bumping to weekly. 
https://github.com/Expensify/App/assets/22508304/ec941317-f911-4f73-9c4e-0c8b06903597

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by mallenexpensify** [comment]
Waiting on retest

**Comment by mallenexpensify** [comment]
Waiting on retest

**Comment by mallenexpensify** [comment]
Waiting on retest

**Comment by mallenexpensify** [comment]
Checking with Isabella about retesting

**Comment by mallenexpensify** [comment]
Was able to reproduce, had to test to understand it (the video confused me for some reason, it looked like the correct number of clicks)
@s77rt can you attempt reproduction and provide any feedback you might have.  Seems like this can be external

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
User is directed to the IOU details page twice when clicking on from link. First to the "[amount] expense", then to the "[user] owes [amount]".
User is forced to click on from link thrice to return to main chat.

### What is the root cause of that problem?
In https://github.com/Expensify/App/blob/83eb7e6ee5f8a86c4500deda3454763680dcd5cf/src/components/ParentNavigationSubtitle.tsx#L44, when going back, we always go back to the `parentReportID`, in this case the `parentReportID` of the thread is the transaction thread.

But the transaction thread is a `oneTransactionThread`, so it's always rendered inside the IOU report itself, it should not be rendered alone. When it's incorrectly rendered alone, clicking back in header will first go back to the IOU report, which again is a one transaction report, so it shows the transaction again. Now click on header of IOU report to go back will go back to main chat.

### What changes do you think we should make in order to solve the problem?

When going back from a thread of the one transaction thread, we should go back to the IOU report because that's where the thread is navigated from.

In https://github.com/Expensify/App/blob/83eb7e6ee5f8a86c4500deda3454763680dcd5cf/src/components/ParentNavigationSubtitle.tsx#L32, add 
```
const [parentReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${parentReportID || -1}`);
const isOneTransactionThread = ReportUtils.isOneTransactionThread(parentReportID, parentReport?.parentReportID ?? '0');
const reportIdToGoBack = isOneTransactionThread ? parentReport?.parentReportID : parentReportID;
```
And in https://github.com/Expensify/App/blob/83eb7e6ee5f8a86c4500deda3454763680dcd5cf/src/components/ParentNavigationSubtitle.tsx#L41-L47
Replace `parentReportID` with `reportIdToGoBack`

[Optional] In `ReportScreen`, check it's a one transaction thread, if correct, navigate to the IOU report. Although with the above change there's no realistic way the user can manually navigate to that one transaction thread

### What alternative solutions did you explore? (Optional)

**Comment by nkdengineer** [comment]
[Proposal updated](https://github.com/Expensify/App/issues/42221#issuecomment-2207851311) to clarify

**Comment by s77rt** [comment]
Reproducible! Technically this is working correctly because the thread level is 3. The report actions we are displaying on the first report (IOU report) are not from that report but from the money request report. Here how it looks originally:


| IOU report<br>2370747844749502 | Money request report<br>772014547788248 |
|:------------------------------:|:----------------------------------------:|
|                  <img width="1725" alt="Screenshot 2024-07-04 at 2 42 04 AM" src="https://github.com/Expensify/App/assets/16493223/81d7d8ae-7728-4cf4-9239-1a951dd66240">              |         <img width="1725" alt="Screenshot 2024-07-04 at 2 42 17 AM" src="https://github.com/Expensify/App/assets/16493223/302feed8-a398-4b67-bf38-ac417381e89b">                                 |

Here how it looks currently (combined report actions)

| IOU report<br>2370747844749502<br>with combined report actions |
|:------------------------------:|
|            <img width="1725" alt="Screenshot 2024-07-04 at 2 46 42 AM" src="https://github.com/Expensify/App/assets/16493223/4a1264ee-b216-4433-b2e0-3ed4bc91d413">                    |

Report levels:
1. IOU report
2. Money request report
3. Date change report

Naturally going from 3 to 1 requires 2 steps. But since the user navigated to 3 from 1 (skipping 2) we should skip that report going back as well. I agree this is `External`

**Comment by s77rt** [comment]
@nkdengineer The RCA is correct. The general idea of the solution makes sense but we still have bugs e.g. going back the report action is not highlighted

https://github.com/Expensify/App/assets/16493223/a94f8d56-7a0b-45db-a077-4c4c5802be41


Manual navigation does not work either, getting not found view

<img width="1725" alt="Screenshot 2024-07-04 at 3 21 25 AM" src="https://github.com/Expensify/App/assets/16493223/7ea92fbb-913e-4200-84ec-df2bb1ee08e8">

**Comment by nkdengineer** [comment]
@s77rt Report action deeplink was never intended to work in one transaction thread. Looking at [this](https://github.com/Expensify/App/blob/main/src/pages/home/ReportScreen.tsx#L380-L385) we can see we try to remove the deeplink to report action right away if both `transactionThreadReportID` exists and `reportActionID` exists in the param.

So it think it should be the author of that code's responsibility to make the reportAction deeplink work for one transaction view, if we decide to allow that.

Regarding why it's showing NotFoundPage for manual navigation, it's because in https://github.com/Expensify/App/blob/58be9bab7d4b82d9895027cdac4501723353c555/src/pages/home/ReportScreen.tsx#L364
It's using `reportActions` which will be empty if `reportActionId` is in the param but not found https://github.com/Expensify/App/blob/58be9bab7d4b82d9895027cdac4501723353c555/src/pages/home/ReportScreen.tsx#L290

To fix it we can use `sortedAllReportActions` [there](https://github.com/Expensify/App/blob/58be9bab7d4b82d9895027cdac4501723353c555/src/pages/home/ReportScreen.tsx#L364), but this is also the scope of "report action deeplinking in one transaction view", I think is out of scope here, as explained above.

**Comment by s77rt** [comment]
@nkdengineer If we make the parent navigation goes directly to the IOU report then we should make the comment linking work correctly. Otherwise we'd be fixing one bug and creating another (inconsistency when going back if we have one vs many expenses)

**Comment by s77rt** [comment]
Not overdue. Still waiting on proposals

**Comment by nkdengineer** [comment]
> @s77rt Report action deeplink was never intended to work in one transaction thread. Looking at [this](https://github.com/Expensify/App/blob/9c32293ab6c0465eec38f1c573b8b5a09e1e112d/src/pages/home/ReportScreen.tsx#L385-L390) we can see we try to remove the deeplink to report action right away if both transactionThreadReportID exists and reportActionID exists in the param.

@s77rt What do you think about this?

I don't think we ever intend for it to work, so we shouldn't try to introduce a feature that we didn't want here.

**Comment by s77rt** [comment]
After checking this more I am inclined into closing this issue. This is not really a bug. The "From" link in the header takes you to the parent report and it's not necessary meant to be used for "going back". In the OP video it looks like they are viewing the same report but it's not really the case.


https://github.com/Expensify/App/assets/16493223/cca83a50-60b3-4529-956e-d85eccb0b303



🎀 👀 🎀 Internal review on the above ^

cc @mallenexpensify @nkdengineer

**Comment by nkdengineer** [comment]
> The "From" link in the header takes you to the parent report and it's not necessary meant to be used for "going back".

As a user, I understand it as "go back to whichever report I come from to this one", in this case "whichever report I come from" is the IOU report and it's making me clicking twice to get back to it, although I go to this thread directly from the IOU report (1 click)

> In the OP video it looks like they are viewing the same report but it's not really the case.

Technically they're 2 reports but, I think from the user's perspective, one-transaction view means that there's only one UI for the transaction. Having 2 different reports with almost same UI and content, and having to navigate through both to get back, is confusing. 

@mallenexpensify @cristipaval  We'd appreciate your inputs here!

**Comment by mallenexpensify** [comment]
> In the OP video it looks like they are viewing the same report but it's not really the case.

Good point, why do they look near-identical? What's the purpose of having two that are near-identical?
<img width="909" alt="image" src="https://github.com/user-attachments/assets/8f3295ca-6377-4bbf-9ec8-f8b08dff08f2">

<img width="835" alt="image" src="https://github.com/user-attachments/assets/221d29f8-24a0-457c-a107-27921e44249b">

**Comment by s77rt** [comment]
@mallenexpensify The first report (in the screenshot) is the transaction report. The second report is the IOU report which is supposed to look like this

<img width="1725" alt="345666969-81d7d8ae-7728-4cf4-9239-1a951dd66240" src="https://github.com/user-attachments/assets/8f9b6ce2-23bf-43d3-a0ae-a11d44862fea">

But instead we are showing the report actions of the transaction report just to save the user from the extra click in the rendered money request preview (https://expensify.slack.com/archives/C01GTK53T8Q/p1720569569451869?thread_ts=1720568655.280159&cid=C01GTK53T8Q)

**Comment by nkdengineer** [comment]
@mallenexpensify FYI currently there's no way to user to navigate directly to the transaction report in case of one-transaction IOU. They will only be able to navigate to the IOU which already contains the information of the transaction. 

If we decide to close this issue, that means the user will be able to navigate to the near-identical transaction report now and I think it's confusing

> save the user from the extra click in the rendered money request preview

As pointed out by @s77rt, we try to save the extra click when navigating to the transaction, I don't see why we don't want to save that extra click when going back (the user will have to click twice across 2 near-identical pages as seen in the OP)

**Comment by s77rt** [comment]
Not overdue. @cristipaval https://github.com/Expensify/App/issues/42221#issuecomment-2218984843. Do you agree on closing this issue?

**Comment by cristipaval** [comment]
yes I am good for closing the issue. Thanks a lot for triaging it 🙏
